mode/smalltalk/smalltalk.js

```diff
@@ -15,6 +15,7 @@ CodeMirror.defineMode('smalltalk', function(config) {
 
   var specialChars = /[+\-\/\\*~<>=@%|&?!.,:;^]/;
   var keywords = /true|false|nil|self|super|thisContext/;
+  var variables = /Smalltalk/;
```

Comment:
You shouldn't use a mode-global variable to keep state -- modes can be restarted and resumed at different points at any time, and thus state should go into the state object.